specutils/io/default_loaders/sdss.py

```diff
@@ -113,38 +159,96 @@ def spSpec_loader(file_obj, **kwargs):
     Returns
     -------
     data: Spectrum1D
-        The spectrum that is represented by the data in this table.
+        The spectrum that is represented by the wavelength solution from the
+        header WCS and data array of the primary HDU.
     """
     if isinstance(file_obj, fits.hdu.hdulist.HDUList):
         hdulist = file_obj
     else:
         hdulist = fits.open(file_obj, **kwargs)
 
     header = hdulist[0].header
-    name = header.get('NAME')
+    # name = header.get('NAME')
     meta = {'header': header}
-    wcs = WCS(hdulist[0].header)
+    wcs = WCS(header).dropaxis(1)
+
+    bunit = header.get('BUNIT', '1e-17 erg / (Angstrom cm2 s)')
+    # fix mutilated flux unit
+    bunit = bunit.replace('/cm/s/Ang', '/ (Angstrom cm2 s)')
+    if 'Ang' in bunit and 'strom' not in bunit:
+        bunit = bunit.replace('Ang', 'Angstrom')
+    flux_unit = Unit(bunit)
+    flux = hdulist[0].data[0, :] * flux_unit
+
+    uncertainty = StdDevUncertainty(hdulist[0].data[2, :] * flux_unit)
```

Comment:
For the record, I double-checked the spSpec data model, and indeed, in those files (only) uncertainty is stored as standard deviation, not inverse variance.

Reply:
Yes, I also verified for the reference spSpec in the tests that its `uncertainty` is indeed 1/sqrt of the corresponding spPlate row (although the test itself does not check this, as it would first have to mask the zeros in the two uncertainties).
The more critical part is the spec data, in which uncertainties had been stored as inverse variance, but were previously converted to standard deviations, because apparently at the time no model for `InverseVariance` existed. But since there is no direct conversion between the two available, it remains to be checked if this would break things downstream.